### PR TITLE
workflow: callback on type/bug issue closed

### LINKED ISSUE #27579: Proposal: Build a lightweight need-more-info logic with concise comments
## Enhancement

Previous discussion at https://internals.tidb.io/t/topic/352

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #27579 

cc @zhouqiang-cl @jebter 

Please review the workflow, and ensure:

1. whether `affects-x.y.z` or `affects-x.y` we want.
2. `needs-more-info` has 1k+ backlog, do you want to squash it or we do a batch omit.
3. @ti-srebot turns off need-more-info logic after this PR merged.

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
LGTM

**Comment by jebter** [comment]
1. Provide branch label by default, for example: affects-4.0, affects-5.0, affects-5.1, affects-5.2. If some bugs only exist in a specific version, provide affects-x.y.z label on demand
2. Please batch omit.

File: .github/workflows/bug-closed.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        uses: andymckay/labeler@1.0.3
+        with:
+          add-labels: "need-more-info"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Add comment
+        uses: peter-evans/create-or-update-comment@v1.4.5
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Please check whether the issue should be labeled with 'affects-x.y.z' or 'backport-x.y.z',
```
Comment (tisonkun) [inline .github/workflows/bug-closed.yml:28]:
```suggestion
            Please check whether the issue should be labeled with 'affects-x.y' or 'backport-x.y.z', 
```

File: .github/workflows/bug-closed.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          add-labels: "need-more-info"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Add comment
+        uses: peter-evans/create-or-update-comment@v1.4.5
+        with:
+          issue-number: ${{ github.event.issue.number }}
+          body: |
+            Please check whether the issue should be labeled with 'affects-x.y' or 'backport-x.y.z', 
+            and then remove 'need-more-info' label.
```
Comment (tisonkun) [inline .github/workflows/bug-closed.yml:29]:
```suggestion
            and then remove 'needs-more-info' label.
```

**Review by zhouqiang-cl — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
Dropped all `need-more-info` backlog.

Use `needs-more-info` to align with other labels (`needs-rebase`, `needs-cheery-pick-`, etc.)

**Review by bb7133 — APPROVED** [review]

**Comment by tisonkun** [comment]
@bb7133  @zhouqiang-cl would you please help on merging this PR？

**Comment by zhouqiang-cl** [comment]
already turn off `need-more-info` logic in ti-srebot

File: .github/workflows/bug-closed.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      !(contains(join(github.event.issue.labels.*.name, ', '), 'affects-') &&
+        contains(join(github.event.issue.labels.*.name, ', '), 'backport-'))
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: Label issues
+        uses: andymckay/labeler@1.0.3
+        with:
+          add-labels: "need-more-info"
```
Comment (tisonkun) [inline .github/workflows/bug-closed.yml:21]:
```suggestion
          add-labels: "needs-more-info"
```

**Pull request merged**